src/ir.cpp

```diff
@@ -11217,11 +11217,45 @@ static IrInstruction *ir_analyze_instruction_add_implicit_return_type(IrAnalyze
     return ir_const_void(ira, &instruction->base);
 }
 
+static bool type_is_error(ZigType *type) {
+    if (!type)
+        return false;
+
+    switch(type->id) {
+        case ZigTypeIdErrorUnion:
+        case ZigTypeIdErrorSet:
+            return true;
+        case ZigTypeIdOptional:
+            return type_is_error(type->data.maybe.child_type);
```

Comment:
Are you sure this case should be here? What would the zig source look like that uses this?